cmd/akamai-purger/main.go

```diff
@@ -181,6 +192,8 @@ func main() {
 		// stop channel. We wait 15 seconds for any remaining URLs to be emptied
 		// from the current queue, if we pass that deadline we exit early.
 		ticker.Stop()
+		// Set queue length gauge on exit.
+		gaugePurgeQueueLength.Set(float64(ap.len()))
```

Comment:
Let's make this a log line instead. We can be confident the log line will get collected, but we can't be sure we'll get scraped by Prometheus in the 15 seconds before exit (it depends on the Prometheus scrape interval)

Reply:
Lines 167 - 177 already handle these bits :)